deploy-lts/docker-compose.yml

```diff
@@ -102,6 +102,31 @@ services:
       test: curl --silent --fail localhost:9999/bigdata/namespace/wdq/sparql
       interval: 10s
       start_period: 2m
+    labels:
+      - "traefik.enable=true"
+      # Define router for wdqs service
+      - "traefik.http.routers.wdqs-router.rule=Host(`${WDQS_PUBLIC_HOST}`)"
+      - "traefik.http.routers.wdqs-router.rule=PathPrefix(`/sparql`)"
+      # Add path prefix
+      - "traefik.http.middlewares.wdqs-prefix.addprefix.prefix=/bigdata/namespace/wdq/"
+      # Enforce HTTP methods
+      - "traefik.http.middlewares.wdqs-headers.headers.accesscontrolallowmethods=GET,OPTIONS,POST"
+      # Headers to enforce read-only and query timeouts
+      - "traefik.http.middlewares.wdqs-headers.headers.customrequestheaders.X-BIGDATA-READ-ONLY=yes"
+      - "traefik.http.middlewares.wdqs-headers.headers.customrequestheaders.X-BIGDATA-MAX-QUERY-MILLIS=300000"
+      # CORS and Vary headers
+      - "traefik.http.middlewares.wdqs-cors-headers.headers.accesscontrolallowheaders=*"
+      - "traefik.http.middlewares.wdqs-cors-headers.headers.addvaryheader=true"
+      # Define wdqs-rate-limit middleware for rate limiting
+      - "traefik.http.middlewares.wdqs-rate-limit.ratelimit.average=10"
+      - "traefik.http.middlewares.wdqs-rate-limit.ratelimit.burst=20"
+      # Apply middlewares to the wdqs router
+      - "traefik.http.routers.wdqs-router.middlewares=wdqs-prefix,wdqs-headers,wdqs-cors-headers,wdqs-rate-limit"
+      # TODO: Set timeouts for the wdqs service
```

Comment:
Is this something still requiring attention?

Reply:
Well, as I now understand, it is basically handled downstream by the `X-BIGDATA-MAX-QUERY-MILLIS` header already. So this would be extra and I would just remove it. Also Traefik `serverstransports` configuration seems cursed to me 👻